### PR TITLE
[EP Perf] MemTest: Add Valgrind and fix addressSanitizer

### PR DESCRIPTION
**Pull request by yf711**
### Description
1. Add valgrind to existing ep_perf CI MemTest and parse ORT-TRT memLeak details
   1. General Valgrind logs and logs related to ORT-TRT will be parsed in [CI artifacts](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=334122&view=artifacts&pathAsName=false&type=publishedArtifacts)
      1. Logic:
         1. Run valgrind with `onnxruntime-perf-test -e tensorrt` and export log to `valgrind.log`
         2. Identify if any `definitely lost` memleak happened
            1. For log paragraphs which show `definitely lost`, parse if they have keyword `TensorrtExecutionProvider`. 
            2. If so, extract these details to `ort_trt_memleak_detail.log`, and return `build failure` to EP Perf CI
3. Fix existing addressSanitizer and sync the squeezenet testcase with latest update from [ort-inference-example](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/c_cxx/squeezenet/main.cpp)
   1. Updates in short: Upgrade main.cpp to be using OrtTensorRTProviderOptionsV2
4. Reorder the 7-min-MemTest to be ahead of 9-hr-model-tests, and enable MemTest by default

---
#### Local testing: `ort_trt_memleak_detail.log` parsed from a `valgrind.log` that showed mem-leaked
 ```bash
100,272 (56 direct, 100,216 indirect) bytes in 1 blocks are definitely lost in loss record 1,751 of 1,832
   at 0x483CFA3: operator new(unsigned long) (vg_replace_malloc.c:472)
   by 0x315DC2: std::_MakeUniq<onnxruntime::OrtAllocatorImplWrappingIAllocator>::__single_object std::make_unique<onnxruntime::OrtAllocatorImplWrappingIAllocator, std::shared_ptr<onnxruntime::IAllocator> >(std::shared_ptr<onnxruntime::IAllocator>&&) (unique_ptr.h:857)
   by 0x30EE7B: OrtApis::KernelContext_GetAllocator(OrtKernelContext const*, OrtMemoryInfo const*, OrtAllocator**) (custom_ops.cc:121)
   by 0x660D115: onnxruntime::TensorrtExecutionProvider::Compile(std::vector<onnxruntime::IExecutionProvider::FusedNodeAndGraph, std::allocator<onnxruntime::IExecutionProvider::FusedNodeAndGraph> > const&, std::vector<onnxruntime::NodeComputeInfo, std::allocator<onnxruntime::NodeComputeInfo> >&)::{lambda(void*, OrtApi const*, OrtKernelContext*)#3}::operator()(void*, OrtApi const*, OrtKernelContext*) const (tensorrt_execution_provider.cc:2223)
```

### CODE REVIEW

**Review by snnn — DISMISSED** [review]

File: tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_8_tensorrt8_6 [hunk 1]
```diff
@@ -41,6 +41,8 @@ RUN v="8.6.1.6-1+cuda11.8" &&\
 # Compile trtexec
 RUN cd /usr/src/tensorrt/samples/trtexec && make
 
+# Install Valgrind
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_8_tensorrt8_6:44]: How about the other docker files? Do you also need to update them?

Comment (yf711) [reply tools/ci_build/github/linux/docker/Dockerfile.ubuntu_cuda11_8_tensorrt8_6:44]: Yes, I will add another commit to update them as well. Thanks for pointing that out!

File: onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
 #include <assert.h>
+#include <iostream>
 #include <vector>
 #include <onnxruntime_cxx_api.h>
 #include <tensorrt_provider_factory.h>
+#include <tensorrt_provider_options.h>
 
-std::unique_ptr<OrtTensorRTProviderOptions> get_default_trt_provider_options() {
-  auto tensorrt_options = std::make_unique<OrtTensorRTProviderOptions>();
+std::unique_ptr<OrtTensorRTProviderOptionsV2> get_default_trt_provider_options() {
```
Comment (jywu-msft) [inline onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp:12]: we aren't supposed to modify the v2 options struct directly. 
we don't need this get_default_trt_provider_options() function anymore after moving to v2 struct. 
you removed the call in line 51.

Comment (yf711) [reply onnxruntime/python/tools/tensorrt/perf/mem_test/main.cpp]: Thanks for pointing that out. This func has been removed as well.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**